Fix frontend-backend integration and improve UX

JonCGroberg opened the pull request:
## Summary
This PR fixes several integration issues between frontend and backend, improves error handling, and enhances the user experience.

## Changes

### Backend Fixes
- **NLP Processing**: Fixed undefined variables in `analyze_texts.py` that were causing 500 errors during document upload
  - Defined `word_frequencies`, `ngrams_data`, `pos_counts_dict`, and `sentence_sentiment` variables
  - Fixed `analysis_results` return value

### Frontend Improvements

#### File Upload
- Removed `user_id` from FormData (backend now correctly extracts from JWT token)
- Improved error parsing to extract FastAPI error details properly
- Added PDF file type validation before upload with user-friendly error messages
- Added file type filtering in upload component (accept .txt only)
- Enhanced drag-and-drop validation

#### UI/UX Enhancements
- **Sentence Display**:
  - Removed sentence limits - now displays all sentences
  - Changed from card-based layout to border-separated list
  - Moved emotion labels to end of sentences for better readability
- **Field Labels**:
  - Updated `vocab_size` label to "unique words" for clarity
  - Updated `word_count` label to "total words" for clarity
- **Aggregated Stats**: Removed sentence limits in aggregated statistics

#### Error Handling
- Better error messages for database connection issues
- Improved token handling in auth context
- Enhanced error handling in API endpoints

### Tests
- Updated test mocks to use `word_count` instead of `token_count` to match backend model

### Documentation
- Updated setup documentation

## Testing
- ✅ Document upload works correctly with JWT authentication
- ✅ PDF files are properly rejected with helpful error messages
- ✅ All sentences are displayed without limits
- ✅ Field labels are clear and accurate
- ✅ Error handling provides user-friendly messages

JonCGroberg committed:
fix(frontend): improve file upload error handling and PDF validation

- Remove user_id from FormData (backend extracts from JWT token)
- Improve error parsing to extract FastAPI error details
- Add PDF file type validation before upload
- Add file type filtering in upload component (accept .txt only)
- Add user-friendly error messages for PDF uploads

JonCGroberg committed:
fix(nlp): define missing variables in analyze_texts for API usage

- Define word_frequencies, ngrams_data, pos_counts_dict, and sentence_sentiment
- Fix analysis_results return value
- Ensure all data structures are properly initialized before use

JonCGroberg committed:
test(frontend): update test mocks to use word_count instead of token_count

JonCGroberg committed:
feat(frontend): improve sentence display and update field labels

- Remove sentence limit - display all sentences
- Change sentence display from cards to border-separated list
- Move emotion labels to end of sentences
- Update vocab_size label to 'unique words'
- Update word_count label to 'total words'
- Remove sentence limits in aggregated stats

JonCGroberg committed:
docs: update setup documentation

JonCGroberg committed:
refactor(auth): improve error handling and user experience

- Better error messages for database connection issues
- Improved token handling in auth context
- Enhanced error handling in API endpoints

lindsayjgc commented:
<img width="1280" height="800" alt="Screenshot 2025-11-16 at 22 52 22" src="https://github.com/user-attachments/assets/609fce88-fae7-453a-83d3-9ce9b8f443e8" />

Receive the above error when I try to 'view analysis' on a document.

JonCGroberg committed:
fix(backend): address security and code quality issues

- Remove unused ssl import from database.py
- Add environment check for TLS certificate validation (only allow invalid certs in development)
- Add textract dependency for PDF and legacy DOC support

JonCGroberg committed:
fix(frontend): improve auth error handling and race conditions

- Fix race condition in AuthContext timeout by tracking initialization completion
- Remove redundant nested try-catch since refreshUser handles errors internally
- Simplify TypeError check in auth.ts to not rely on error message content
- Change NodeJS.Timeout to number type for browser compatibility

JonCGroberg committed:
fix(frontend): fix response body stream consumption issue

- Read response text first, then parse as JSON to avoid consuming body stream twice
- Improves error message extraction from FastAPI error responses

JonCGroberg committed:
fix(frontend): address UI/UX review comments

- Fix indentation issue in document detail page
- Simplify UploadCard file filtering logic and add warning for filtered files
- Restore PDF/DOC file support (backend handles conversion)
- Add comment explaining defense-in-depth validation approach

JonCGroberg committed:
fix(frontend): use ReturnType for setTimeout to handle Node.js and browser types

- Change timeoutId type from number to ReturnType<typeof setTimeout>
- Fixes TypeScript build error where setTimeout returns Timeout in Node.js but number in browser

JonCGroberg committed:
fix(frontend): fix all linting errors and warnings

- Remove unused imports (Progress, BookOpen, useAuth, Separator)
- Fix unused variables by prefixing with underscore
- Replace array index keys with unique identifiers (ngram strings, emotion names)
- Fix accessibility issues in UploadCard (add role and keyboard handlers)
- Fix React hooks dependencies (wrap refreshUser in useCallback)
- Remove redundant error handling (try-catch blocks that only rethrow)
- Replace non-null assertions with optional chaining
- Replace global isNaN/isFinite with Number.isNaN/Number.isFinite
- Fix unused function parameters
- Auto-format code with Biome

JonCGroberg committed:
fix(frontend): prevent double file selection in UploadCard

- Stop event propagation on button click to prevent div onClick from firing
- Fixes issue where users had to select files twice due to event bubbling

lindsayjgc reviewed (approved):
looks good

lindsayjgc committed:
Merge branch 'dev' into fix/frontend-backend-integration-improvements

lindsayjgc committed:
fixed typo caused by resolving conflicts
